### PR TITLE
Remove enzyme

### PR DESCRIPTION
**Pull request by vsumner**
## Description
This PR removes `enzyme` from quilt along with deprecating:
- `jest-mock-apollo`
- `jest-mock-router`
- `enzyme-utilities`

These packages are outdated and no longer maintained. I have removed the contents and left the readme so that we can publish empty versions and remove the directories in a following PR.

Fixes (issue #)

## Type of change
I changed some tests to remove enzyme. I didn't change the actual packages so I'm not including a changelog entry.

### CODE REVIEW

File: tests/consistent-changelogs.test.ts [hunk 1]
```diff
@@ -111,7 +111,12 @@ function readChangelogs() {
   const packagesPath = join(ROOT_PATH, 'packages');
 
   return glob
-    .sync(join(packagesPath, '*/'))
+    .sync(
+      join(
+        packagesPath,
+        '!(jest-mock-apollo|enzyme-utilities|jest-mock-router)/',
```
Comment (vsumner) [inline tests/consistent-changelogs.test.ts:117]: These are temporary and can be removed once the empty packages are published and then removed.

**Review by BPScott — APPROVED** [review]
I wonder if we could offer a "and here's what you should used instead" guidance in the new readmes, but otherwise yay 🔥 🔥 🔥

File: packages/enzyme-utilities/README.md [hunk 2]
```diff
@@ -1,61 +1,5 @@
...
-    const myComponent = mount(<MyComponent />);
-    const pause = trigger(myComponent.find(Button), 'onAction');
-    expect(myComponent.find(Thing)).toHaveLength(0);
-
-    await pause;
-    expect(myComponent.find(Thing)).toHaveLength(1);
-  });
-});
-```
+This package is no longer supported and has been deprecated.
```
[the remark this answers is not in the document]
Comment (vsumner) [reply packages/enzyme-utilities/README.md:5]: Good point. I can do that tomorrow.

[the remark this answers is not in the document]
Comment (vsumner) [reply packages/enzyme-utilities/README.md]: I added recommendations for `enzyme-utilities` and `jest-mock-apollo`. We don't have a replacement for `jest-mock-router` which only mocked `react-router` 3.

**Comment by BPScott** [comment]
:shipit:

**Pull request merged**